client/ayon_core/tools/sceneinventory/model.py

```diff
@@ -126,32 +131,41 @@ def refresh(self, selected=None):
         """Refresh the model"""
         # for debugging or testing, injecting items from outside
         container_items = self._controller.get_container_items()
-
         self._clear_items()
-
-        items_by_repre_id = {}
+        repre_id = set()
```

Comment:
Whatever will be using this ids further down wouldn't know about which project the `repre_id` belongs to - introducing potentially subtle bugs. It may be good to add a comment here now and create an issue where relevant OR fix it now. Since this potentially touches site sync I may be worth adding todo comment here now and make an issue on ayon-sitesync to orchestrate fixing that down the line.

Also, this holds representation ids, so please make it plural.

```suggestion
        repre_ids = set()
```